prescient_sdk/upload.py

```diff
@@ -54,6 +54,23 @@ def _upload(
     s3.upload_file(Filename=file, Bucket=bucket, Key=key)
 
 
+def _make_s3_key(file: Path, root: Path) -> str:
+    """
+    Compute an S3 key for `file` relative to the `root` directory.
+
+    Ensures the key is normalized with forward slashes (POSIX style),
+    so it works consistently across Windows and POSIX systems.
+
+    Args:
+        file (Path): The full path to the file being uploaded.
+        root (Path): The root input directory passed to `upload`.
+
+    Returns:
+        str: The normalized S3 key.
+    """
+    return file.relative_to(root).as_posix()
```

Comment:
this works, however from my testing I observe that the files uploaded to the bucket do not contain the parent dir as part of the key e.g.

`input_dir="/home/ysebastien/test"`

with a file contained such as `/home/ysebastien/test/data.tif` should result in a s3 key `s3://upload-bucket/test/data.tif` but currently it would be `s3://upload-bucket/data.tif`.